named_arrays/_vectors/vectors.py

```diff
@@ -85,6 +81,38 @@ def type_matrix(self) -> Type[na.AbstractExplicitMatrixArray]:
         The corresponding :class:`named_arrays.AbstractMatrixArray` class
         """
 
+    @property
+    def matrix(self) -> na.AbstractMatrixArray:
+        new_dict = {}
+        for c in self.components:
+            component = self.components[c]
+            if isinstance(component, AbstractVectorArray):
+                new_dict[c] = component.matrix
+            elif isinstance(component, na.AbstractMatrixArray):
+                raise NotImplemented
```

Comment:
You can't raise `NotImplemented`, it has to be `NotImplementedError`

Reply:
I would say that it would be better if this looked like (tabs are wrong
```
if isinstance(component, AbstractVectorArray):
    new_dict[c] = component.matrix
elif isinstance(na.as_named_array(component), na.AbstractScalar)
    new_dict[c] = component
else:
    raise NotImplementedError
```
That way we're not making as many assumptions about what the subclasses of vector are

Reply:
Good idea.  Fixed